tox.ini

```diff
@@ -0,0 +1,59 @@
+# This configuration file is for Tox. https://tox.readthedocs.io/
+# It houses configuration sections for common Python tools, too.
+
+[tox]
+envlist =
+    flake8
+    pylint
+    readme
+    py{27,33,34}-django17
+    py{27,33,34,35}-django18
+    py{27,34,35}-django19
+    py{27,34,35}-django110
+    py{27,34,35,36}-django111
+    py{34,35,36}-django20
+
+[testenv]
+commands =
+    flake8: flake8
+    pylint: pylint --rcfile=tox.ini pylint_django setup
+    readme: python setup.py check --restructuredtext --strict
+    py{27,34,35,36}-django{17,18,19,110,111,20}: coverage run test/test_func.py
+    clean: find . -type f -name '*.pyc' -delete
+    clean: find . -type d -name __pycache__ -delete
+    clean: rm -rf build/ .cache/ dist/ .eggs/ pylint_django.egg-info/ .tox/
+deps =
+    flake8: flake8
+    pylint: pylint
+    pylint: Django>=2.0,<2.1
+    readme: readme_renderer
+    django17: Django>=1.7,<1.8
+    django18: Django>=1.8,<1.9
+    django19: Django>=1.9,<1.10
+    django110: Django>=1.10,<1.11
+    django111: Django>=1.11,<2.0
+    django20: Django>=2.0,<2.1
+    py{27,34,35,36}: coverage
+    py{27,34,35,36}: djangorestframework
+    py{27,34,35,36}: psycopg2
+    py{27,34,35,36}: django-tables2
+    py{27,34,35,36}: pylint-plugin-utils
+    py{27,34,35,36}: pytest
+setenv =
+    PIP_DISABLE_PIP_VERSION_CHECK = 1
+    PYTHONPATH = .
+whitelist_externals =
+    clean: find
+    clean: rm
+
+[travis:env]
+DJANGO =
+    1.7: django17
+    1.8: django18
+    1.9: django19
+    1.10: django110
+    1.11: django111
+    2.0: django20
```

Comment:
Hmmm, this I don't like much. It is duplicating what we have in .travis.yml. Can you instead of listing these mappings in tox.ini use the env variables provided from Travis ? Maybe have a generic tox target to run the tests and install the dependencies but use whatever Django version has been specified from the outside ?

Reply:
You shouldn't worry about the numbers here. This is the way we have to do it when using `tox-travis`. See the [discussion in their repo](https://github.com/tox-dev/tox-travis/issues/77).

The actual dependencies are [defined in tox.ini](https://github.com/PyCQA/pylint-django/pull/126/files#diff-b91f3d5bd63fcd17221b267e851608e8R30), by the way. This is the most generic and readable way to date, according to my knowledge. If you know any better, please suggest an example.